fix: dark mode persistence on Learn page

yashsaraswat2004 opened the pull request:
## Description
This PR fixes #182  the dark mode persistence issue on the Learn page. The theme now persists across page navigation and reloads.

## Changes
- Replaced hardcoded colors with CSS variables.
- Saved theme preference in `localStorage`.
- Applied the saved theme on page load by setting the `data-theme` attribute.

## Related Issue


## Screenshots
### Before
![Before](
![Screenshot (217)](https://github.com/user-attachments/assets/9d2d4b96-f3fa-4c37-acee-3905fe29ceb7)
)

### After
![After](
![Screenshot (219)](https://github.com/user-attachments/assets/251582d6-d141-4a3b-999e-1285620460c6)
)

## Checklist
- [x] Replaced hardcoded colors with CSS variables.
- [x] Saved theme preference in `localStorage`.
- [x] Applied the saved theme on page load.
- [x] Tested changes across multiple browsers.
- [x] Ensured all unit tests pass.

## Notes for Reviewers
- Please verify the dark mode behavior on the Learn page.
- Ensure the theme persists across page reloads and navigation.

DianaLease commented on src/components/ToggleDarkMode.tsx:
```diff
@@ -14,6 +14,9 @@ const ToggleDarkMode: React.FC = () => {
     setIsDarkMode((prev) => !prev);
+    const newTheme = !isDarkMode ? "dark" : "light";
+    document.documentElement.setAttribute("data-theme", newTheme);
+    console.log('ToggleDarkMode: new isDarkMode:', !isDarkMode, 'data-theme set to:', newTheme, 'localStorage:', localStorage.getItem('darkMode'));
```
Remove console.log

DianaLease commented on src/pages/LearnNow.tsx:
```diff
@@ -1,14 +1,23 @@
 const LearnNow: React.FC = () => {
+  useEffect(() => {
+    const computedBg = getComputedStyle(document.documentElement).getPropertyValue('--bg-color');
+    console.log("LearnNow: computed --bg-color:", computedBg);
```
Remove console.log

DianaLease commented on src/components/Navbar.tsx:
```diff
@@ -85,19 +93,78 @@ function Navbar({ scrollToFooter }: { scrollToFooter: any }) {
     </Menu>
   );
 
+  const mobileMenu = (
```
These navbar menu changes are unrelated to the dark mode on the Learn page. Can you remove the unrelated changes and keep this PR focused around only the dark mode issue?

yashsaraswat2004 replied:
I see your point! I've reverted the navbar changes and kept this PR focused only on the dark mode updates. Also I've removed the console.log statements as requested .Thanks for the feedback!

yashsaraswat2004 commented:
Hey @DianaLease . I've made the requested changes: removed console.log statements and reverted unrelated navbar modifications. Let me know if any further tweaks are needed. Thanks for the review!

DianaLease commented:
> Hey @DianaLease . I've made the requested changes: removed console.log statements and reverted unrelated navbar modifications. Let me know if any further tweaks are needed. Thanks for the review!

Thanks! Can you merge the latest main into your branch?

yashsaraswat2004 commented:
I have merged the latest main into my branch. Thanks @DianaLease!"

DianaLease commented:
Thanks for the contribution!
